src/App.tsx

```diff
@@ -2,49 +1,109 @@
 import 'bulma/css/bulma.css';
+import { useState } from 'react';
+import classNames from 'classnames';
+
 import './App.scss';
+import { goodsFromServer } from './api/goodsFromServer';
+import { SortType } from './enums/sortingEnum';
+
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoods = (compareFunction: (a: string, b: string) => number) => {
+    setGoods(prevGoods => [...prevGoods].sort(compareFunction));
+  };
+
+  const handleSort = (typeOfSortGoods: SortType) => {
+    let compareFunction;
+
+    switch (typeOfSortGoods) {
+      case SortType.Alphabetically:
+        compareFunction = (a: string, b: string) => {
+          return isReversed ? b.localeCompare(a) : a.localeCompare(b);
+        };
+
+        setActiveSort(SortType.Alphabetically);
+        break;
+
+      case SortType.ByLength:
+        compareFunction = (a: string, b: string) => {
+          return isReversed ? b.length - a.length : a.length - b.length;
+        };
+
+        setActiveSort(SortType.ByLength);
+        break;
+
+      default:
+        throw new Error(`Unknown sort type: ${typeOfSortGoods}`);
+    }
+
+    sortGoods(compareFunction);
+  };
+
+  const handleReverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevIsReversed => !prevIsReversed);
+  };
+
+  const handleResetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort('');
+  };
```

Comment:
[CODE STYLE] - The function's name 'handleResetGoods' should start with a verb indicating its action. Consider renaming it to 'resetGoodsToDefault' to be more descriptive.